app/libs/twitter_processor.rb

```diff
@@ -0,0 +1,43 @@
+require "twitter"
+
+class TwitterProcessor
+  def initialize(company:, number_of_tweets:)
+    @twitter_link = company.twitter
+    @count = number_of_tweets
+    @client = initialize_twitter_client
+  end
+
+  def screen_name
+    @twitter_link.sub('https://twitter.com/', '') unless @twitter_link.blank?
+  end
+
+  def tweets
+    screen_name.blank? ? [] : scrape_tweets
```

Comment:
cache it, using 

```
@tweets ||= begin
  CODE
end
```

because you calling tweets multiple times

and similar for other places